src/Agent.Listener/JobDispatcher.cs

```diff
@@ -602,6 +605,34 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, WorkerDisp
                                 var messageType = MessageType.CancelRequest;
                                 if (HostContext.AgentShutdownToken.IsCancellationRequested)
                                 {
+                                    var service = HostContext.GetService<IFeatureFlagProvider>();
+                                    var ffState = await service.GetFeatureFlagAsync(HostContext, "DistributedTask.Agent.FailJobWhenAgentDies", Trace);
+                                    if (ffState.EffectiveState == "On")
+                                    {
```

Comment:
[not blocking] this feels like it should be in a single function (perhaps in another class) rather than inline